meta/views.py

```diff
@@ -2045,10 +2045,11 @@ def taxa_page(request):
 
 def places_page(request):
     '''Página mostrando locais de maneira organizada.'''
-    locations = Location.objects.order_by('name')
-    cities = City.objects.order_by('name')
-    states = State.objects.order_by('name')
-    countries = Country.objects.order_by('name')
+    locations = Location.objects.filter(id__in=Media.objects.values('location')).order_by('name')
```

Comment:
For locations, cities, states, and countries, one can use the same `related_name` query as below:

```
locations = Location.objects.filter(media__isnull=False).distinct()
```

Reply:
Using `exclude` is the best:

```python
Location.objects.exclude(media__isnull=True)
```

Reply:
OK! Changed using exclude.